Tasks/Nikiforov/Task2/Task2.py

```diff
@@ -0,0 +1,65 @@
+import re
+
+d = {0:[[], ["ноль", "первого", "час"]], 1:[["одна"], ["один", "второго", "два"]], 2:[["две"], ["два", "третьего", "три"]], 
+3:[["три"], ["три", "четвертого", "четыре"]], 4:[["четыре"], ["четыре", "пятого", "пять"]], 5:[["пять"], ["пять", "шестого", "шесть"]],
+6:[["шесть"], ["шесть", "седьмого", "семь"]], 7:[["семь"], ["семь", "восьмого", "восемь"]], 8:[["восемь"], ["восемь", "девятого", "девять"]], 
+9:[["девять"], ["девять", "десятого", "десять"]], 10:[["десять"], ["десять", "одиннадцатого", "одиннадцать"]], 
+11:[["одиннадцать"], ["одиннадцать", "двенадцатого", "двенадцать"]], 12:[["двенадцать"], ["двенадцать", "первого", "час"]],
+13:[["тринадцать"]], 14:[["четырнадцать"]], 15:[["пятнадцать"]], 16:[["шестнадцать"]], 17:[["семнадцать"]], 18:[["восемнадцать"]],
+19:[["девятнадцать"]], 20:[["двадцать"]], 21:[["двадцать одна"]], 22:[["двадцать две"]], 23:[["двадцать три"]], 24:[["двадцать четыре"]],
+25:[["двадцать пять"]], 26:[["двадцать шесть"]], 27:[["двадцать семь"]], 28:[["двадцать восемь"]], 29:[["двадцать девять"]], 30:[["тридцать"]],
+31:[["тридцать одна"]], 32:[["тридцать две"]], 33:[["тридцать три"]], 34:[["тридцать четыре"]], 35:[["тридцать пять"]], 36:[["тридцать шесть"]],
+37:[["тридцать семь"]], 38:[["тридцать восемь"]], 39:[["тридцать девять"]], 40:[["сорок"]], 41:[["сорок одна"]], 42:[["сорок две"]],
+43:[["сорок три"]], 44:[["сорок четыре"]], 45:[["пятнадцати"]], 46:[["четырнадцати"]], 47:[["тринадцати"]], 48:[["двенадцати"]],
+49:[["одиннадцати"]], 50:[["десяти"]], 51:[["девяти"]], 52:[["восьми"]], 53:[["семи"]], 54:[["шести"]],
+55:[["пяти"]], 56:[["четырех"]], 57:[["трех"]], 58:[["двух"]], 59:[["одной"]]}  
+          
+time = input("Введите время в формате hh:mm: \n")
+time = time.replace(" ", "").replace(".",":").replace(",",":")
+
+check = '^([0-9]|([0,1][0-9])|(2[0-3])):[0-9]|[0-5][0-9]$'
```

Comment:
re is a good instrument, but it tends to be overcomplicated, I would use something like r'^[0-9]{2}:[0-5][0-9]$' and then perform some additional validation against an hour value, but it's just preference I think